app/models/spree/order_decorator.rb

```diff
@@ -82,6 +83,20 @@ def products_available_from_new_distribution
     errors.add(:base, I18n.t(:spree_order_availability_error)) unless DistributionChangeValidator.new(self).can_change_to_distribution?(distributor, order_cycle)
   end
 
+  def using_guest_checkout?
+    require_email && !user.andand.id
+  end
+
+  def registered_email?
+    Spree.user_class.find_by_email(email).present?
```

Comment:
[exists?](https://apidock.com/rails/v3.2.13/ActiveRecord/FinderMethods/exists%3F) is a bit more lightweight and exactly meant for things like this.

Reply:
Good point, I will quickly change that.